zcash_client_sqlite/src/wallet/init/migrations/add_transaction_views.rs

```diff
@@ -186,17 +196,25 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
                           ON transactions.id_tx = received_notes.tx
                    LEFT JOIN blocks
                           ON transactions.block = blocks.height
-            GROUP BY received_notes.tx;
-            CREATE VIEW v_transactions AS
-            SELECT id_tx,
-                   mined_height,
-                   tx_index,
-                   txid,
-                   expiry_height,
-                   raw,
-                   SUM(value) + MAX(fee) AS net_value,
-                   SUM(is_change) > 0 AS has_change,
-                   SUM(memo_present) AS memo_count
+            GROUP BY received_notes.tx, received_notes.account;",
+        )?;
+
+        transaction.execute_batch(
+            "CREATE VIEW v_transactions AS
+            SELECT notes.id_tx,
+                   notes.mined_height,
+                   notes.tx_index,
+                   notes.txid,
+                   notes.expiry_height,
+                   notes.raw,
+                   SUM(notes.value) + MAX(notes.fee) AS net_value,
+                   MAX(notes.fee)                    AS fee_paid,
+                   SUM(notes.is_wallet_internal) > 0 AS is_wallet_internal,
```

Comment:
`is_wallet_internal` is maybe misleading at a transaction level; this will be set to `true` for a regular user-created transaction that has a change output. If that's the desired effect then this should be renamed to `has_wallet_internal`. However, I think the desired check here is that _every_ output is wallet-internal, so this should probably be: 
```suggestion
                   SUM(notes.is_not_wallet_internal) == 0 AS is_wallet_internal,
```

Reply:
Good suggestion; I did this as `is_outgoing` instead.